### PR TITLE
Implemented changes for set chef extension client_pem option to support validator less bootstrap support 

### PR DESCRIPTION
**Pull request by siddheshwar-more**
User can specify already created node relevant chef client key details using this option. This will avoid use of organization validation pem file. 

Note- Before use of this options user should create client on chef server for node and specify corresponding client pem using this option. 

Next step: Add support in existing chef extension commands to trigger chef server api and create/register client to support validator less bootstrap

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@siddheshwar-more__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, AZPRBOT;

**Comment by siddheshwar-more** [comment]
Tests recording pending - Facing some issue and working on same

**Comment by siddheshwar-more** [comment]
@yugangw-msft 

Resolved merge conflict and also did recording for tests. Ready for merge!

Thanks!

File: test/data/set-chef-extension-client.pem [hunk 1]
```diff
@@ -0,0 +1,23 @@
+-----BEGIN CERTIFICATE-----
+MIID6zCCAtOgAwIBAgIJAIqaAKIUFOEuMA0GCSqGSIb3DQEBBQUAMIGLMQswCQYD
```
Comment (yugangw-msft) [inline test/data/set-chef-extension-client.pem]: Want to confirm this is a faked or expired cert file, right?

File: lib/commands/asm/vm/vmclient.js [hunk 2]
```diff
@@ -3024,9 +3033,14 @@ _.extend(VMClient.prototype, {
                 }
               }
               options[propTo] = JSON.stringify(config);
-            } else if (propTo == 'privateConfig') {
-              config['validation_key'] = data.toString();
-              options[propTo] = JSON.stringify(config);
+            } else if (propTo == 'privateConfig' ) {
```
Comment (yugangw-msft) [inline lib/commands/asm/vm/vmclient.js]: suggest use '==='

**Comment by yugangw-msft** [comment]
CI passed: https://travis-ci.org/Azure/azure-xplat-cli/builds/88796901
@huangpf , please review

**Comment by huangpf** [comment]
@yugangw-msft LGTM

File: test/data/set-chef-extension-client.pem (hunk 1 above)
Comment (siddheshwar-more) [inline test/data/set-chef-extension-client.pem]: yeah, its a faked file!

File: lib/commands/asm/vm/vmclient.js (hunk 2 above)
Comment (siddheshwar-more) [inline lib/commands/asm/vm/vmclient.js]: @yugangw-msft -
Fixed review comments. Thank You!

**Pull request merged**